Makefile

```diff
@@ -223,52 +185,20 @@ push-multi-arch-node-manager-image: push-all-node-images ## Create and push a ma
 	done
 	docker manifest push --purge $(NODE_MANAGER_IMAGE)
 
-.PHONY: push-all-node-images ## Push node-manager image for os and archs.
-push-all-node-images: push-all-node-images-linux push-all-node-images-windows
-
-.PHONY: push-all-node-images-linux ## Push node-manager image for Linux.
-push-all-node-images-linux: $(addprefix push-node-image-linux-,$(ALL_ARCH.linux))
-
-.PHONY: push-all-node-images-windows ## Push node-manager image for Windows.
-push-all-node-images-windows: $(addprefix push-node-image-windows-,$(ALL_OS_ARCH.windows))
-
-# split words on hyphen, access by 1-index
-word-hyphen = $(word $2,$(subst -, ,$1))
+.PHONY: build-all-images
+build-all-images: buildx-setup
+	ENABLE_GIT_COMMAND="$(ENABLE_GIT_COMMAND)" VERSION="$(VERSION)" OUTPUT="$(OUTPUT_TYPE)" IMAGE_TAG="$(IMAGE_TAG)" docker buildx bake --pull  --progress plain -f buildx-bake.hcl 
 
-push-node-image-linux-%:
-	$(MAKE) ARCH=$* push-node-image-linux
-
-push-node-image-windows-%:
-	$(MAKE) WINDOWS_OSVERSION=$(call word-hyphen,$*,1) ARCH=$(call word-hyphen,$*,2) OUTPUT_TYPE=registry build-node-image-windows
-
-.PHONY: build-all-node-images ## Build node-manager image for all OS and archs.
-build-all-node-images: build-all-node-images-linux build-all-node-images-windows
-
-.PHONY: build-all-node-images-linux ## Build node-manager image for Linux.
-build-all-node-images-linux: $(addprefix build-node-image-linux-,$(ALL_ARCH.linux))
-
-.PHONY: build-all-node-images-windows ## Build node-manager image for Windows.
-build-all-node-images-windows: $(addprefix build-node-image-windows-,$(ALL_OS_ARCH.windows))
-
-build-node-image-linux-%:
-	$(MAKE) ARCH=$* build-node-image-linux
-
-build-node-image-windows-%:
-	$(MAKE) WINDOWS_OSVERSION=$(call word-hyphen,$*,1) ARCH=$(call word-hyphen,$*,2) build-node-image-windows
-
-.PHONY: build-all-ccm-images
-build-all-ccm-images: $(addprefix build-ccm-image-,$(ALL_ARCH.linux))
-
-build-ccm-image-%:
-	$(MAKE) ARCH=$* build-ccm-image
-
-.PHONY: push-all-ccm-images
-push-all-ccm-images: $(addprefix push-ccm-image-,$(ALL_ARCH.linux))
-
-push-ccm-image-%:
-	$(MAKE) ARCH=$* push-ccm-image
+build-ccm-image-%: buildx-setup
+	ENABLE_GIT_COMMAND="$(ENABLE_GIT_COMMAND)" VERSION="$(VERSION)" OUTPUT="$(OUTPUT_TYPE)" IMAGE_TAG="$(IMAGE_TAG)" docker buildx bake --pull  --progress plain -f buildx-bake.hcl ccm-$*
 
+.PHONY: build-ccm-image
+build-ccm-image: buildx-setup build-ccm-image-amd64 ## Build controller-manager image.
 
+.PHONY: build-node-image-linux
+build-node-image-linux: buildx-setup 
```

Comment:
`OUTPUT_TYPE="docker,dest=$(image.release.dir)/linux/$(arch)/azure-cloud-node-manager.tar" ARCH=$(arch) make build-node-image-linux` is used to build cnm with specific arch, can we support this?

https://dev.azure.com/AzureContainerUpstream/Kubernetes/_git/ado-build-definitions?path=/pipelines/cloud-provider.azure.build.yml&version=GBmaster&line=53&lineEnd=54&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

Reply:
IMAGE_REGISTRY=martindlut OUTPUT=registry make build-node-image-linux can be used to build image for all platform.

Reply:
Is it possible to build arm64 cnm image alone? Through the  buildx baker target you provided `"cnm-linux-amd64","cnm-linux-arm64","cnm-linux-arm",`

Reply:
try build-node-image-linux